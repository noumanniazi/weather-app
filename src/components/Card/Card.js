// lib
import React from "react";
import PropTypes from "prop-types";
// Smart imports reduces the build size, instead of building the whole
// library it builds with only specific components
import AntDCard from "antd/lib/card";

// src
import { formatDate, DATE_FORMAT } from '../../utils';
import "./Card.scss";

const Card = props => {
  const { temperature, date, humidity, handleCardClick, isSelected } = props;
  return (
    <div
      className={`cardWrapper ${isSelected === true ? "selected" : ""}`}
      onClick={() => {
        handleCardClick(date);
      }}
    >
      <AntDCard hoverable>
        <div>
          <span className="bold-text">Temp: </span>
          <span>{temperature}</span>
        </div>
        <div>
          <span className="bold-text">Date: </span>
          <span>{formatDate(date, DATE_FORMAT)}</span>
        </div>
        <div>
          <span className="bold-text">humidity: </span>
          <span>{humidity}</span>
        </div>
      </AntDCard>
    </div>
  );
};

Card.prototype = {
  temperature: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  humidity: PropTypes.string.isRequired,
  handleCardClick: PropTypes.func.isRequired,
  isSelected: PropTypes.bool.isRequired
};
export default Card;
