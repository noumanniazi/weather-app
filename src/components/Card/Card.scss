.cardWrapper {
  display: flex;
  width: 33%;
  &.selected {
    .ant-card {
      &.ant-card-bordered {
        border: 2px solid black;
      }
    }
  }
  .ant-card {
    margin: 10px;
    @media (max-width: 40em) {
      margin: 5px;
      span{
        font-size: 12px;
      }
    }
    width: 100%;
    cursor: pointer;
    .bold-text {
      font-weight: bold;
    }
    .ant-card-body {
      @media (max-width: 40em) {
        padding: 3px;
      }
    }
  }
}
