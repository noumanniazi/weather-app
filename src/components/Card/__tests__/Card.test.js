// lib
import React from "react";
import renderer from "react-test-renderer";

// src
import Card from "../Card";

describe("Testing Card Component", () => {
  it("Matches Card snapshot", () => {
    const data = {
      temperature: "6.13C",
      humidity: 81.75,
      date: "2019-12-18",
      isSelected: false
    };
    const card = renderer.create(<Card {...data} />);
    let tree = card.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
