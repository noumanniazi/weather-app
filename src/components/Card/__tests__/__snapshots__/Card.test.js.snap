// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card Component Matches Card snapshot 1`] = `
<div
  className="cardWrapper "
  onClick={[Function]}
>
  <div
    className="ant-card ant-card-bordered ant-card-hoverable"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div>
        <span
          className="bold-text"
        >
          Temp: 
        </span>
        <span>
          6.13C
        </span>
      </div>
      <div>
        <span
          className="bold-text"
        >
          Date: 
        </span>
        <span>
          2019-12-18
        </span>
      </div>
      <div>
        <span
          className="bold-text"
        >
          humidity: 
        </span>
        <span>
          81.75
        </span>
      </div>
    </div>
  </div>
</div>
`;
